### PR TITLE
Add ASP.NET Core section to demo DI

### PR DESCRIPTION
**Pull request by lilyjma**
Context: 

Since Dependency Injection is a key workflow for .NET developers and we've seen requests from customers ([twitter](https://twitter.com/WestDiscGolf/status/1387084607493914627), [stackoverflow](https://stackoverflow.com/questions/52206269/azure-search-net-sdk-searchindexclient-dependency-injection)) requesting for DI related content for the new SDKs, we're adding a section about DI for ASP.NET Core in the readme. This has been discussed various folks (David Fowler, David Pine, Scott Addie etc.), and they think it'll be helpful to users. 

cc @maggiepint

### CODE REVIEW

File: sdk/search/Azure.Search.Documents/README.md [hunk 1]
```diff
@@ -96,6 +96,33 @@ string key = Environment.GetEnvironmentVariable("SEARCH_API_KEY");
...
+    services.AddControllers();
+}
+```
+To use the preceding code, add this to your configuration:
+
+```json
+{
+    "Search": {
+      "endpoint": "https://<resource-name>.search.windows.net",
+      "credential": { "key": "resource key" },
```
Comment (brjohnstmsft) [inline sdk/search/Azure.Search.Documents/README.md:119]: Is there a way to achieve this without hardcoding the API key in a config file? Or is "resource key" the name of an environment variable...?

Comment (lilyjma) [reply sdk/search/Azure.Search.Documents/README.md:119]: It's not the name of the environment variable. It is the key of the resource. @pakrym is the way to do what Bruce is asking?

Comment (pakrym) [reply sdk/search/Azure.Search.Documents/README.md:119]: The IConfiguration system uses many sources of configuration, in ASP.NET Core one of them is environment variables. You can define the same key as SEARCH__CREDENTIAL__KEY="..." environment variable (the __ is used as a section separator).

Comment (brjohnstmsft) [reply sdk/search/Azure.Search.Documents/README.md:119]: Thanks for the context. Can we clarify the example to make it clear that it follows that pattern? We don't want to encourage customers to hard-code secrets in config.

Comment (JoshLove-msft) [reply sdk/search/Azure.Search.Documents/README.md:119]: I wonder if this feedback should apply to the other libraries as well, i.e. should we apply this to the Service Bus readme also?

Comment (lilyjma) [reply sdk/search/Azure.Search.Documents/README.md:119]: @joshfree I'm adding this to all the libraries that have the extension method for DI (some libraries don't from my investigation. I'll create issues for them.) I've added the section for SB already :)

Comment (JoshLove-msft) [reply sdk/search/Azure.Search.Documents/README.md:119]: But the Service Bus readme uses the connection string in the config. In this Readme, it sounds like we are going to be updating it to not include secrets in the config. That is what I am wondering about.

**Comment by lilyjma** [comment]
@pakrym and @brjohnstmsft: sorry for the delay. I've made the changes. Please take a look.

File: sdk/search/Azure.Search.Documents/README.md [hunk 2]
```diff
@@ -96,6 +96,47 @@ string key = Environment.GetEnvironmentVariable("SEARCH_API_KEY");
...
+You'll also need to provide your resource key to authenticate the client, but you shouldn't be putting that information in the configuration. Instead, when in development, use [User-Secrets](https://docs.microsoft.com/aspnet/core/security/app-secrets?view=aspnetcore-5.0&tabs=windows#how-the-secret-manager-tool-works). Add the following to `secrets.json`:
+
+```json
+{
+    "SearchClient": {
+      "credential": { "key": "<you resource key>" }
+    }
+}
+```
+When running in production, it's preferable to use [environment variables](https://docs.microsoft.com/aspnet/core/security/app-secrets?view=aspnetcore-5.0&tabs=windows#environment-variables):
```
Comment (pakrym) [inline sdk/search/Azure.Search.Documents/README.md:132]: We might want to add this information to docs.microsoft.com/en-us/dotnet/azure/sdk/dependency-injection#store-configuration-separately-from-code as well.

File: sdk/search/Azure.Search.Documents/README.md [hunk 3]
```diff
@@ -96,6 +96,47 @@ string key = Environment.GetEnvironmentVariable("SEARCH_API_KEY");
...
+
+```json
+{
+    "SearchClient": {
+      "endpoint": "https://<resource-name>.search.windows.net",
+      "indexname": "nycjobs"
+    }
+}
+```
+You'll also need to provide your resource key to authenticate the client, but you shouldn't be putting that information in the configuration. Instead, when in development, use [User-Secrets](https://docs.microsoft.com/aspnet/core/security/app-secrets?view=aspnetcore-5.0&tabs=windows#how-the-secret-manager-tool-works). Add the following to `secrets.json`:
```
Comment (brjohnstmsft) [inline sdk/search/Azure.Search.Documents/README.md:123]: Is "resource key" common terminology across the Azure SDK? In Search REST API docs, we call these "API keys".

**Review by bitsforthought — APPROVED** [review]

**Pull request merged**